dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconController.java

```diff
@@ -126,17 +128,58 @@ public void getIconData(@PathVariable String key, HttpServletResponse response)
   @GetMapping
   public @ResponseBody List<IconResponse> getAllIcons(
       @RequestParam(required = false) String[] keywords) {
+
+    IconCriteria iconCriteria = IconCriteria.of(0, 0, true);
+
     List<Icon> icons;
 
     if (keywords == null) {
-      icons = iconService.getIcons();
+      icons = iconService.getIcons(iconCriteria);
     } else {
       icons = iconService.getIcons(keywords);
     }
 
     return icons.stream().map(iconMapper::from).toList();
   }
 
+  @GetMapping(value = "/custom")
+  public @ResponseBody PagingWrapper<IconResponse> getAllCustomIcons(
+      @RequestParam(defaultValue = "1") int page,
+      @RequestParam(defaultValue = "5") int size,
+      @RequestParam(defaultValue = "false") boolean skipPaging,
+      @RequestParam(required = false) String[] keywords) {
+
+    IconCriteria iconCriteria = IconCriteria.of(page, size, skipPaging);
+
+    List<CustomIcon> icons;
+
+    if (keywords == null) {
+      icons = iconService.getCustomIcons(iconCriteria);
+    } else {
+      icons = iconService.getCustomIcons(keywords);
+    }
+
+    int count = iconService.getIconCount();
+
+    List<IconResponse> iconResponses = icons.stream().map(iconMapper::from).toList();
+
+    PagingWrapper<IconResponse> listPagingWrapper = new PagingWrapper<>();
```

Comment:
This is the pager used in Tracker and we shouldn't use it.
We should use something like here https://github.com/dhis2/dhis2-core/blob/master/dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/AbstractFullReadOnlyController.java#L199-L269

Anyway, this pagination issue will be tackled in tracker in https://dhis2.atlassian.net/browse/TECH-1670, maybe you can even leave as it is now and we can go back here when we fix Tracker so we will have a common solution for it

Reply:
this has been in use outside the tracker as well.
`ProgramNotificationInstanceController
`DeduplicationController. 
Secondly, the custom icon does not seem to fall in the metadata domain.

Reply:
I agree to have common solution. There has been many ways we can do paging in dhis2-core. We should have a uniform way for pagination implementation.